data/items.ts

```diff
@@ -66,7 +69,16 @@ export const Items: {[itemid: string]: ItemData} = {
 		fling: {
 			basePower: 30,
 		},
+		onBoostPriority: 1,
+		onBoost(boost, target) {
+			target.itemState.lastAtk = target.boosts['atk'];
+		},
 		onAfterBoost(boost, target, source, effect) {
+			const noAtkChange = boost.atk && boost.atk < 0 && target.boosts['atk'] === -6 && target.itemState.lastAtk === -6;
+			const noContraryAtkChange = boost.atk && boost.atk > 0 && target.boosts['atk'] === 6 && target.itemState.lastAtk === 6;
+			if (target.boosts['spe'] === 6 || noAtkChange || (target.hasAbility('contrary') && noContraryAtkChange)) {
```

Comment:
No need to check for Contrary here, the only way Intimidate could have raised your Attack is if you had Contrary.

Reply:
i just wanted to make sure defiant didnt mess anything up here

Reply:
If you have Defiant then you couldn't have been at +6 before and after when the boost was positive.